lib/msf/core/payload_generator.rb

```diff
@@ -314,20 +317,25 @@ def generate_payload
       if platform == "java" or arch == "java" or payload.start_with? "java/"
         raw_payload = generate_java_payload
         cli_print "Payload size: #{raw_payload.length} bytes"
-        raw_payload
+        gen_payload = raw_payload
       elsif payload.start_with? "android/" and not template.blank?
         cli_print "Using APK template: #{template}"
         apk_backdoor = ::Msf::Payload::Apk::ApkBackdoor::new()
         raw_payload = apk_backdoor.backdoor_apk(template, generate_raw_payload)
         cli_print "Payload size: #{raw_payload.length} bytes"
-        raw_payload
+        gen_payload = raw_payload
       else
         raw_payload = generate_raw_payload
         raw_payload = add_shellcode(raw_payload)
         encoded_payload = encode_payload(raw_payload)
         encoded_payload = prepend_nops(encoded_payload)
         cli_print "Payload size: #{encoded_payload.length} bytes"
-        format_payload(encoded_payload)
+        gen_payload = format_payload(encoded_payload)
+      end
+      if gen_payload.length > @space and not @smallest
```

Comment:
this errors out in an uncaught way if gen_payload is nil. That happens when an issue like #6915 occurs (wrong format for the payload).

Suggest a patch like this and kill 2 birds with one stone:
```
diff --git a/lib/msf/core/payload_generator.rb b/lib/msf/core/payload_generator.rb
index 987871c..ec8c96b 100644
--- a/lib/msf/core/payload_generator.rb
+++ b/lib/msf/core/payload_generator.rb
@@ -332,7 +332,9 @@ module Msf
         cli_print "Payload size: #{encoded_payload.length} bytes"
         gen_payload = format_payload(encoded_payload)
       end
-      if gen_payload.length > @space and not @smallest
+      if gen_payload.nil?
+        raise PayloadGeneratorError, 'The payload could not be generated, check options'
+      elsif gen_payload.length > @space and not @smallest
         raise PayloadSpaceViolation, 'The payload exceeds the specified space'
       else
         gen_payload
```